service/iot-service-samples/pnp-service-sample/TemperatureController/src/main/java/samples/com/microsoft/azure/sdk/iot/TemperatureController.java

```diff
@@ -0,0 +1,100 @@
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.PnpHelper;
+import com.microsoft.azure.sdk.iot.service.devicetwin.*;
+import com.microsoft.azure.sdk.iot.service.exceptions.IotHubException;
+
+import java.io.IOException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.TimeUnit;
+
+// This sample uses the model - https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/TemperatureController.json.
+public class TemperatureController {
+    // Get connection string and device id inputs.
+    private static final String iotHubConnectionString  = System.getenv("IOTHUB_CONNECTION_STRING");
+    private static final String deviceId = System.getenv("DEVICE_ID");
+
+    private static DeviceTwin twinClient;
+    private static DeviceMethod methodClient;
+
+    public static void main(String[] args) throws Exception {
+        RunSample();
+    }
+
+    private static void RunSample() throws IOException, IotHubException {
+        System.out.println("Initialize the service client.");
+        InitializeServiceClient();
+
+        System.out.println("Get Twin model Id and Update Twin");
+        GetAndUpdateTwin();
+
+        System.out.println("Invoke a method on component");
+        InvokeMethodOnComponent();
+
+        System.out.println("Invoke a method on root level");
+        InvokeMethodOnRootLevel();
+    }
+
+    private static void InitializeServiceClient() throws IOException {
+        twinClient = DeviceTwin.createFromConnectionString(iotHubConnectionString);
+        methodClient = DeviceMethod.createFromConnectionString(iotHubConnectionString);
+    }
+
+    private static void GetAndUpdateTwin() throws IOException, IotHubException {
+        // Get the twin and retrieve model Id set by Device client.
+        DeviceTwinDevice twin = new DeviceTwinDevice(deviceId);
+        twinClient.getTwin(twin);
+        System.out.println("Model Id of this Twin is: " + twin.getModelId());
+
+        // Update the twin for thermostat1 component.
+        System.out.println("Updating Device twin property");
+        String propertyName = "targetTemperature";
+        String propertyValue = "60";
```

Comment:
I was wondering if this should be on top. That's the approach I took in C#.
Don't have a strong opinion though

Reply:
I have kept it as is for now. We can see later which way to go.